Mathlib/Computability/Language.lean

```diff
@@ -228,15 +230,17 @@ theorem add_supr {ι : Sort v} [Nonempty ι] (l : ι → Language α) (m : Langu
 #align language.add_supr Language.add_supr
 
 theorem mem_pow {l : Language α} {x : List α} {n : ℕ} :
-    x ∈ l ^ n ↔ ∃ S : List (List α), x = S.join ∧ S.length = n ∧ ∀ y ∈ S, y ∈ l :=
-  by
+    x ∈ l ^ n ↔ ∃ S : List (List α), x = S.join ∧ S.length = n ∧ ∀ y ∈ S, y ∈ l := by
   induction' n with n ihn generalizing x
   · simp only [mem_one, pow_zero, length_eq_zero]
     constructor
     · rintro rfl
-      exact ⟨[], rfl, rfl, fun y h => h.elim⟩
-    · rintro ⟨_, rfl, rfl, _⟩
-      rfl
+      exact ⟨[], rfl, rfl, λ _ h ↦ by contradiction⟩
+    · intro h; have ⟨ax, bx, cx, dx⟩ := h
+      rw [bx]; simp; intros l' h'
+      rw [length_eq_zero] at cx
+      rw [cx] at h'
+      contradiction
```

Comment:
Why can't you do a recursive `intro`? That's what the proof used to be.
Please leave a `-- Porting note:` that records the old proof, if you can't get it to work.

Reply:
@jcommelin Thanks for the comments!